### ISSUE TITLE
CEPL doesn't generate uniform uploaders for structs correctly

### ISSUE THREAD

**Issue by djeis97** [cbaggers/cepl#230]
```lisp
(defun-g foo-vert (&uniform (v g-pnt))
  (pos v))

(defun-g foo-frag ()
  (vec4 1 0 0 1))
```

This code has a few problems with CEPL as it stands, despite it being legal GLSL. The first seems to be an artifact of a refactor to varjo- it tries to call `varjo.internals:type-spec->type` on the slot types, despite varjo already storing those as type objects. The second is that struct uploads seem to always pass through CFFI, which causes problems for structs with non-primitive types like `:vec4`.

**Comment by djeis97** [comment]
Also reading the code it looks like it generates the uniform names from the varjo slot name, while varjo currently generates the glsl slot names from the accessor name.

**Comment by cbaggers** [comment]
Thanks man, the good news is that I have struct refactorings landing now. 
I'll try this and see if it's fixed (if not I'll try fix it tonight)
I'll also make a test for this and get it into [cepl.tests](https://github.com/cbaggers/cepl.tests) so it shouldn't happen again

**Comment by cbaggers** [comment]
Can confirm its not fixed by the first batch of tests. Will look at this next

**Comment by cbaggers** [comment]
Should be fixed by 703a4c9f6cd9895703bb091bc1df2cec071efef8. The test for this can be found here: https://github.com/cbaggers/cepl.tests/blob/master/tests/structs.lisp#L57

Gonna run through the release tests now, if we can I'd like to get this fix out in the January release

Thanks again @djeis97, your info was spot on

**Comment by cbaggers** [comment]
tentatively closing this. Please reopen if you still see issues on master
